Status: sent-upstream
# Latest updates from 2010-04-12 for fbset.8 not sent.

---
 fb.modes.5    |   38 +++++++++++++++++++++++++++++++++++++-
 fbset.8       |   41 ++++++++++++++++++++++++++++++-----------
 modeline2fb.1 |   40 ++++++++++++++++++++++++++++++++++++++++
 3 files changed, 107 insertions(+), 12 deletions(-)

--- a/fb.modes.5
+++ b/fb.modes.5
@@ -1,4 +1,4 @@
-.TH fb.modes 8 "Aug 1996" local "Linux frame buffer utils"
+.TH fb.modes 5 2003-08-07 2.1 "Linux frame buffer utils"
 .SH NAME
 fb.modes \- frame buffer modes file
 .SH DESCRIPTION
@@ -27,6 +27,9 @@ timings
 .br
 .B options
 .RI < value >
+.br
+.B rgba
+.RI < red , green , blue , alpha >
 .RE
 endmode
 .SH OPTIONS
@@ -74,6 +77,22 @@ horizontal sync length (in pixels)
 vertical sync length (in pixel lines)
 .RE
 .PP
+rgba options (only valid with truecolor):
+.RS
+.TP
+.I red
+red color bitfields (in length or length/offset)
+.TP
+.I green
+green color bitfields (in length or length/offset)
+.TP
+.I blue
+blue color bitfields (in length or length/offset)
+.TP
+.I alpha
+alpha color bitfields (in length or length/offset)
+.RE
+.PP
 other options:
 .RS
 the first value of this options is the default
@@ -87,6 +106,14 @@ the vertical sync polarity
 .IR \fBcsync "\ {" low | high }
 the composite sync polarity
 .TP
+.IR \fBgsync "\ {" low | high }
+the sync on green polarity
+.TP
+.IR \fBbcast "\ {" false | true }
+enable or disable broadcast modes. If enabled the frame buffer generates
+the exact timings fot several broadcast modes (e.g. PAL or NTSC). Note that
+this option may not be supported by every frame buffer
+.TP
 .IR \fBextsync "\ {" false | true }
 enable or disable external resync. If enabled the sync timings are not
 generated by the frame buffer device and must be provided externally
@@ -106,6 +133,15 @@ and this way the horizontal frequency ca
 same resolution can be displayed on different monitors, even if the
 horizontal frequency specification differs. Note that this option may not be
 supported by every frame buffer device
+.TP
+.IR \fBnostd "\ <" number >
+select nonstandard video mode
+.TP
+.IR \fBaccel "\ {" false | true }
+enable or disable hardware text acceleration
+.TP
+.IR \fBgrayscale "\ {" false | true }
+enable or disable graylevels instead of colors
 .RE
 .SH INTERNALS
 Generally a frame buffer display is organized as follows:
--- a/fbset.8
+++ b/fbset.8
@@ -1,4 +1,4 @@
-.TH fbset 8 "July 1998" local "Linux frame buffer utils"
+.TH fbset 1 2010-04-12 2.1 "Linux frame buffer utils"
 .SH NAME
 fbset \- show and modify frame buffer device settings
 .SH SYNOPSIS
@@ -6,8 +6,6 @@ fbset \- show and modify frame buffer de
 .RI [ options ]
 .RI [ mode ]
 .SH DESCRIPTION
-.B This documentation is out of date!!
-.PP
 .B fbset
 is a system utility to show or change the settings of the frame buffer
 device. The frame buffer device provides a simple and unique interface to
@@ -36,10 +36,8 @@ General options:
 .BR \-\-help ",\ " \-h
 display an usage information
 .TP
-.BR \-\-now ",\ " \-n
-change the video mode immediately. If no frame buffer device is given via
-.B \-fb
-, then this option is activated by default
+.BR \-\-test
+don't change, just test whether the mode is valid
 .TP
 .BR \-\-show ",\ " \-s
 display the video mode settings. This is default if no further option or
@@ -66,14 +64,16 @@ display the timing information as it's n
 Frame buffer device nodes:
 .RS
 .TP
+.BR \-\-all ",\ "  \-a
+change all virtual consoles on this device
+.TP
 .BR \-fb "\ <" \fIdevice >
 .I device
 gives the frame buffer device node. If no device via
 .B \-fb
 is given, 
 .I /dev/fb0
 is used
-.TP
 .RE
 .PP
 Video mode database:
@@ -86,6 +88,13 @@ see also
 .BR fb.modes (5)
 .RE
 .PP
+Display bitfield colors:
+.RS
+.TP
+.BR \-rgba "\ <" \fIred , \fIgreen , \fIblue , \fIalpha >
+each in length or length/offset color format
+.RE
+.PP
 Display geometry:
 .RS
 .TP
@@ -104,6 +113,10 @@ set virtual vertical resolution (in pixe
 .BR \-depth "\ <" \fIvalue >
 set display depth (in bits per pixel)
 .TP
+.TP
+.BR \-nonstd "\ <" \fIvalue >
+select nonstandard video mode
+.TP
 .BR \-\-geometry ",\ " \-g "\ ..."
 set all geometry parameters at once in the order
 .RI < xres >
@@ -154,13 +167,16 @@ set all timing parameters at once in the
 .RI < hslen >
 .RI < vslen >,
 e.g.
-.B \-g
+.B \-t
 .I 35242 64 96 35 12 112 2
 .RE
 .PP
 Display flags:
 .RS
 .TP
+.IR \fB\-accel "\ {" false | true }
+set hardware text acceleration enable
+.TP
 .IR \fB\-hsync "\ {" low | high }
 set the horizontal sync polarity
 .TP
@@ -170,6 +186,9 @@ set the vertical sync polarity
 .IR \fB\-csync "\ {" low | high }
 set the composite sync polarity
 .TP
+.IR \fB\-gsync "\ {" false | true }
+set synch on green
+.TP
 .IR \fB\-extsync "\ {" false | true }
 enable or disable external resync. If enabled the sync timings are not
 generated by the frame buffer device and must be provided externally
@@ -217,7 +236,7 @@ insert the following in rc.local:
 .B fbset
 -fb
 .I /dev/fb0
-vga
+640x480\-60
 .sp
 .RE
 and make the used frame buffer device known to
@@ -232,7 +251,7 @@ and make the used frame buffer device kn
 .br
 .I /etc/fb.modes
 .SH SEE ALSO
-.BR fb.modes "(5), " fbdev (4)
+.BR fb.modes "(5), " fbdev "(4), " /usr/share/doc/fbset/FAQ.gz
 .SH AUTHORS
 .TP
 Geert Uytterhoeven <Geert.Uytterhoeven@cs.kuleuven.ac.be>
--- /dev/null
+++ b/modeline2fb.1
@@ -0,0 +1,40 @@
+.TH modeline2fb 1 2006-01-18 2.1 "Linux frame buffer utils"
+.SH NAME
+modeline2fb \- simple modeline to fb.modes translator
+.SH SYNOPSIS
+.B modeline2fb
+[\fIOPTION\fR] [\fIFILES\fR]
+.SH DESCRIPTION
+.PP
+.I Modeline2fb
+is a simple Perl script that converts XF86Config-style modelines to options
+suitable for a fb.modes file.
+.PP
+Note that only one option can be successfully enabled at any particular time.
+.SH OPTIONS
+.TP
+\fB\-d\fR, \fB\-\-depth\fR \fIdepth\fR
+Use the given display depth (default is 8).
+.TP
+\fB\-h\fR \fB\-\-help\fR
+Print out a help screen and exit.
+.SH ADVANCED OPTIONS
+.TP
+\fB\-r\fR \fB\-\-rounding\fR \fIdiv\fR
+Sets the vxres divisor (default is 128).
+.TP
+\fB\-x\fR \fB\-\-vxres\fR \fIX,X,X,...\fR
+Sets extra vxres values.
+.PP
+[\fIFILES\fR] refers to one or more XF86Config files.  Note that all modelines 
+must be in single-line format.  If no files are given on the command line, 
+this program reads from standard in.  This program will also write to
+standard out.
+.SH EXAMPLE 
+modeline2fb \-d 16 /etc/X11/XF86Config
+.SH "SEE ALSO"
+.BR fb.modes(5), 
+.BR XF86Config(5)
+.SH AUTHOR
+This manual page is a quick write-up for Debian done by Kevin Kreamer 
+<kkreamer@etherhogz.org>.
